script_library/cleanup-k8s.sh

```diff
@@ -16,17 +26,35 @@ rm -rf /var/lib/nova/*
 rm -rf /etc/libvirt/qemu/*
 findmnt --raw | awk '/^\/var\/lib\/kubelet\/pods/ { print $1 }' | xargs -r -L1 -P16 sudo umount -f -l
 
-for pvc in $(kubectl get pvc -n openstack | awk ' NR > 1 { print $1 ; }'); do
-    kubectl delete pvc $pvc -n openstack --all;
+deleted_items=()
+echo "deleting persistent volume claims in namespaces: ${deploy_namespaces[@]}"
+for NS in "${deploy_namespaces[@]}"; do
+    for pvc in $(kubectl get pvc -n $NS | awk ' NR > 1 { print $1 ; }'); do
+        kubectl delete pvc $pvc -n $NS;
+        deleted_items+=("$NS:$pvc")
+    done
 done
-
-for pv in $(kubectl get pv -n openstack | awk ' NR > 1 { print $1 ; }'); do
-    kubectl delete pv $pv -n openstack --all || true;
+echo "deleted persistent volume claims : ${deleted_items[@]}"
+
+deleted_items=()
+echo "deleting persistent volumes in namespaces: ${deploy_namespaces[@]}"
+for NS in "${deploy_namespaces[@]}"; do
+    for pv in $(kubectl get pv -n $NS | awk ' NR > 1 { print $1 ; }'); do
+        kubectl delete pv $pv -n $NS || true;
```

Comment:
I was not in favour of the --all in the first place, so I am fine with removing it.